diff --git a/ion/src/simulator/linux/Makefile b/ion/src/simulator/linux/Makefile
--- a/ion/src/simulator/linux/Makefile
+++ b/ion/src/simulator/linux/Makefile
@@ -21,15 +21,20 @@
 ion_src += $(addprefix ion/src/simulator/shared/, \
   dummy/haptics_enabled.cpp \
   dummy/keyboard_callback.cpp \
   dummy/window_callback.cpp \
   actions.cpp \
   clipboard_helper.cpp \
-  collect_registers_x86_64.s \
   collect_registers.cpp \
   haptics.cpp \
   journal.cpp \
   state_file.cpp \
   store_script.cpp \
 )
 
+ifeq ($(shell uname -m),aarch64)
+ion_src += ion/src/simulator/shared/collect_registers_arm64.s
+else
+ion_src += ion/src/simulator/shared/collect_registers_x86_64.s
+endif
+
 ifeq ($(EPSILON_TELEMETRY),1)
