diff --git a/hw/arm/fsl-imx6ul.c b/hw/arm/fsl-imx6ul.c
--- a/hw/arm/fsl-imx6ul.c
+++ b/hw/arm/fsl-imx6ul.c
@@ -214,2 +214,7 @@
-    create_unimplemented_device("a7mpcore-mmdc", FSL_IMX6UL_MMDC_CFG_ADDR,
-                                FSL_IMX6UL_MMDC_CFG_SIZE);
+    {
+        DeviceState *mmdc = qdev_new(TYPE_PRIME_G2_MMDC);
+        object_property_add_child(OBJECT(dev), "mmdc", OBJECT(mmdc));
+        sysbus_realize_and_unref(SYS_BUS_DEVICE(mmdc), &error_abort);
+        sysbus_mmio_map(SYS_BUS_DEVICE(mmdc), 0, FSL_IMX6UL_MMDC_CFG_ADDR);
+    }
